addons/account/models/account_account.py

```diff
@@ -635,20 +592,46 @@ def load(self, fields, data):
         with opening debit/credit. In that case, the auto-balance is postpone
         until the whole file has been imported.
         """
-        rslt = super(AccountAccount, self).load(fields, data)
+        if 'opening_balance' not in fields:
+            return super().load(fields, data)
+
+        # Collect the opening_balance. It's compute/inverse field so nothing happen and the field will be empty at
+        # the end of the import.
+        super_set_opening_balance = type(self.env['account.account'])._set_opening_balance
+        opening_balance_per_account = {}
+
+        def set_opening_balance(records):
+            super_set_opening_balance(records)
+            for record in records:
+                opening_balance_per_account[record] = record.opening_balance
+
+        with patch.object(type(self.env['account.account']), '_set_opening_balance', set_opening_balance):
+            rslt = super().load(fields, data)
```

Comment:
@william-andre Because the inverse will check the move, create a line, then auto-balance the move and it's slowing down the whole import. What we had before was a weird context key to bypass the auto-balancing. However, the rest was still not optimal.
I'm not proud of this part :/ I only have the raw data as parameter and I don't have any helper to parse it according what the import does. Instead of reinventing the wheel, I used the patch function to keep the regular decoding data flow.

Reply:
> since it's done account per account.

It doesn't have to be the case